File: frontend/src/lib/tco/presets.ts
import { CloudCostAssumptions, LocalCostAssumptions } from "./models";

export interface ScenarioPreset {
  name: string;
  description: string;
  cloud: CloudCostAssumptions;
  local: LocalCostAssumptions;
}

export const PRESETS: Record<string, ScenarioPreset> = {
  "Steady VM estate": {
    name: "Steady VM estate",
    description: "Balanced compute and storage spend.",
    cloud: { computeMonthly: 12000, storageMonthly: 4000, databaseMonthly: 2500, networkMonthly: 1000, backupMonthly: 750, supportMonthly: 1500, annualGrowthRatePct: 12, discountPct: 0 },
    local: { infrastructureSubscriptionMonthly: 10000, softwareLicensesMonthly: 1200, supportContractMonthly: 1300, powerCoolingMonthly: 900, datacenterMonthly: 600, adminLaborMonthly: 3000, backupDrMonthly: 1000, annualGrowthRatePct: 3, migrationOneTime: 35000, installationOneTime: 5000 },
  },
  "Storage-heavy archive": {
    name: "Storage-heavy archive",
    description: "Large and growing storage footprint.",
    cloud: { computeMonthly: 5000, storageMonthly: 14000, databaseMonthly: 1000, networkMonthly: 2500, backupMonthly: 2000, supportMonthly: 1500, annualGrowthRatePct: 18, discountPct: 5 },
    local: { infrastructureSubscriptionMonthly: 13500, softwareLicensesMonthly: 1000, supportContractMonthly: 1800, powerCoolingMonthly: 1300, datacenterMonthly: 900, adminLaborMonthly: 3500, backupDrMonthly: 1750, annualGrowthRatePct: 4, migrationOneTime: 55000, installationOneTime: 8000 },
  },
  "Compute-heavy platform": {
    name: "Compute-heavy platform",
    description: "High monthly compute run rate.",
    cloud: { computeMonthly: 28000, storageMonthly: 3500, databaseMonthly: 5500, networkMonthly: 1200, backupMonthly: 1000, supportMonthly: 2400, annualGrowthRatePct: 10, discountPct: 15 },
    local: { infrastructureSubscriptionMonthly: 20000, softwareLicensesMonthly: 2800, supportContractMonthly: 2500, powerCoolingMonthly: 1800, datacenterMonthly: 1200, adminLaborMonthly: 5000, backupDrMonthly: 1500, annualGrowthRatePct: 3, migrationOneTime: 70000, installationOneTime: 12000 },
  },
};

File: src/tco/presets.py
from __future__ import annotations

from dataclasses import dataclass

from .models import CloudCostAssumptions, LocalCostAssumptions


@dataclass(frozen=True)
class ScenarioPreset:
    name: str
    description: str
    cloud: CloudCostAssumptions
    local: LocalCostAssumptions


PRESETS: dict[str, ScenarioPreset] = {
    "Steady VM estate": ScenarioPreset(
        name="Steady VM estate",
        description="Balanced compute and storage spend with moderate cloud growth.",
        cloud=CloudCostAssumptions(
            compute_monthly=12_000,
            storage_monthly=4_000,
            database_monthly=2_500,
            network_monthly=1_000,
            backup_monthly=750,
            support_monthly=1_500,
            annual_growth_rate_pct=12,
            discount_pct=0,
        ),
        local=LocalCostAssumptions(
            infrastructure_subscription_monthly=10_000,
            software_licenses_monthly=1_200,
            support_contract_monthly=1_300,
            power_cooling_monthly=900,
            datacenter_monthly=600,
            admin_labor_monthly=3_000,
            backup_dr_monthly=1_000,
            annual_growth_rate_pct=3,
            migration_one_time=35_000,
            installation_one_time=5_000,
        ),
    ),
    "Storage-heavy archive": ScenarioPreset(
        name="Storage-heavy archive",
        description="Large and growing storage footprint with lower compute intensity.",
        cloud=CloudCostAssumptions(
            compute_monthly=5_000,
            storage_monthly=14_000,
            database_monthly=1_000,
            network_monthly=2_500,
            backup_monthly=2_000,
            support_monthly=1_500,
            annual_growth_rate_pct=18,
            discount_pct=5,
        ),
        local=LocalCostAssumptions(
            infrastructure_subscription_monthly=13_500,
            software_licenses_monthly=1_000,
            support_contract_monthly=1_800,
            power_cooling_monthly=1_300,
            datacenter_monthly=900,
            admin_labor_monthly=3_500,
            backup_dr_monthly=1_750,
            annual_growth_rate_pct=4,
            migration_one_time=55_000,
            installation_one_time=8_000,
        ),
    ),
    "Compute-heavy platform": ScenarioPreset(
        name="Compute-heavy platform",
        description="High monthly compute run rate with meaningful committed-use opportunity.",
        cloud=CloudCostAssumptions(
            compute_monthly=28_000,
            storage_monthly=3_500,
            database_monthly=5_500,
            network_monthly=1_200,
            backup_monthly=1_000,
            support_monthly=2_400,
            annual_growth_rate_pct=10,
            discount_pct=15,
        ),
        local=LocalCostAssumptions(
            infrastructure_subscription_monthly=20_000,
            software_licenses_monthly=2_800,
            support_contract_monthly=2_500,
            power_cooling_monthly=1_800,
            datacenter_monthly=1_200,
            admin_labor_monthly=5_000,
            backup_dr_monthly=1_500,
            annual_growth_rate_pct=3,
            migration_one_time=70_000,
            installation_one_time=12_000,
        ),
    ),
}

